lib/typecheck/ty_env.ml

```diff
@@ -97,6 +97,9 @@ let join : Interface_env.t -> t -> t -> Position.t -> t * Constraint_set.t =
                               pattern = Some pat;
                               quasilinearity = ql
                           }, constrs
+                | List ty1, List ty2 ->
```

Comment:
This is my omission, but can you please add in cases for Tuples and Sums following how it's done in `intersect_types`?